src/import/sst_service.rs

```diff
@@ -675,6 +684,62 @@ impl<E: Engine> ImportSstService<E> {
     }
 }
 
+fn check_local_region_stale(
+    region_id: u64,
+    epoch: &RegionEpoch,
+    local_region_info: Option<RegionInfo>,
+) -> Result<()> {
+    match local_region_info {
+        Some(local_region_info) => {
+            let local_region_epoch = local_region_info.region.region_epoch.unwrap();
+
+            // when local region epoch is stale, client can retry write later
+            if is_epoch_stale(&local_region_epoch, epoch) {
+                let request_region_epoch = RegionEpoch {
+                    conf_ver: epoch.conf_ver,
+                    version: epoch.version,
+                    unknown_fields: Default::default(),
+                    cached_size: Default::default(),
+                };
+                return Err(Error::RequestTooNew {
+                    region_id,
+                    local_region_epoch,
+                    request_region_epoch,
+                });
+            }
+            // when local region epoch is ahead, client need to rescan region from PD to get
+            // latest region later
+            if is_epoch_stale(epoch, &local_region_epoch) {
+                let request_region_epoch = RegionEpoch {
+                    conf_ver: epoch.conf_ver,
+                    version: epoch.version,
+                    unknown_fields: Default::default(),
+                    cached_size: Default::default(),
+                };
+                return Err(Error::RequestTooOld {
+                    region_id,
+                    local_region_epoch,
+                    request_region_epoch,
+                });
+            }
+
+            // not match means to rescan
+            Ok(())
+        }
+        None => {
+            // when region not found, we can't tell whether it's stale or ahead, so we just
+            // return the safest case
+            Err(Error::Engine(
```

Comment:
https://github.com/tikv/tikv/pull/15795/commits/8bb6a380d572884493d5bcdb5321cf3428e45d3f I prefer to use two error enum because they have different error message templates and client will check the substring of the message.